AutoRefAPI.md

```diff
@@ -0,0 +1,187 @@
+It is used to communicate to the AutoRef whether the command that was 
+received was legal in the current game state.
+For every message received, the GameController will send a response message:
+
+``<id>:OK`` In case the command was valid and the operation has been
+performed
+
+``<id>:ILLEGAL`` In case the command was valid but the operation 
+was rejected by the GameController due to preconditions not being met
+(e.g., the current game state does not allow the operation to be 
+performed)
+
+``<id>:INVALID`` In case an unknown command was sent to the GameController.
+
+In the following, all valid message types with the set of parameters 
+will be defined:
+
+### Kick-Off
+
+Format: ``<id>:KICKOFF:<team_id>``
+
+The GameController automatically sets the team playing on the left
+side to have kick-off. Before the game starts, the team having kick-off
+can be explicitely set using this message.
+
+The ``team_id`` is the unique identifier of a team that needs to match the ID indicated in the
+``game.json`` config.
+
+### Playing Sides
+
+Format: ``<id>:SIDE_LEFT:<team_id>``
+
+The GameController automatically sets the sides a team is playing on
+based on the team colors. However, this can be manually changed using 
+this message. The message sets explicitly which team plays on the 
+left side of the field. The right side is adjusted accordingly.
+
+The ``team_id`` is the unique identifier of a team that needs to match the ID indicated in the
+``game.json`` config.
+
+
+### Clock
+
+Format: ``<id>:CLOCK:<time_in_ms>``
+
+The GameControllerSimulator requires the time to be kept and communicated
+via `CLOCK` messages. It is not set up to work with the system's
+internal clock. As parameter, it expects the current time in milliseconds.
+
+The GameController can run in realtime as well as faster or slower than
+realtime. 
+
+### State
+
+Format: ``<id>:STATE:<gamestate>``
+
+Accepted States: ``READY|SET|PLAY|FINISH``
+
+The GameState is used to explicitly set the current state of the game
+in the beginning of a half time, after a game was scored and when the
+game time has exceeded the length of a half time.
+
+### Penalty
+
+Format: ``<id>:PENALTY:<team_id>:<robot_id>:<Offense>``
+
+Accepted Offenses: ``BALL_MANIPULATION|PICKUP|INCAPABLE|PHYSICAL_CONTACT``
+
+The Penalties are used to communicate that a robot is removed from the field
+suffering a 30 sec removal penalty. The ``team_id`` is the unique
+identifier of a team that needs to match the ID indicated in the 
+``game.json`` config. The ``robot_id`` is the identifier of the robot
+(between 1 and 4 in KidSize and 1 and 2 in AdultSize).
+
+Important: This is not used for direct or indirect free kicks! They
+require a different format!
+
+### Score
+
+Format: ``<id>:SCORE:<team_id>``
+
+This message indicates that a goal was scored by a given team.
+The ``team_id`` is the unique identifier of a team that needs to match the ID indicated in the
+``game.json`` config.
+
```

Comment:
Maybe add an example trace of a couple commands to eg. start a game

Reply:
Sample to start a game is already provided in the sim_client/main.py -> I added this to the documentation